gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/repositories/specifications/DocumentPredicates.java

```diff
@@ -2,36 +2,73 @@
 
 import com.querydsl.core.types.ExpressionUtils;
 import com.querydsl.core.types.Predicate;
+import com.querydsl.core.util.StringUtils;
+import com.querydsl.jpa.JPAExpressions;
 import dev.ctrlspace.gendox.gendoxcoreapi.model.QDocumentInstance;
+import dev.ctrlspace.gendox.gendoxcoreapi.model.QProjectDocument;
 import dev.ctrlspace.gendox.gendoxcoreapi.model.dtos.criteria.DocumentCriteria;
 
 
-
+import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 
 public class DocumentPredicates {
 
-    private static QDocumentInstance documentInstance = QDocumentInstance.documentInstance;
+    private static QDocumentInstance qDocumentInstance = QDocumentInstance.documentInstance;
+    private static QProjectDocument qProjectDocument = QProjectDocument.projectDocument;
+
 
-    public static Predicate build(DocumentCriteria criteria){
+    public static Predicate build(DocumentCriteria criteria) {
         return ExpressionUtils.allOf(
-                organizationId(criteria.getOrganizationId())
-               // userId(criteria.getUserId())
+                organizationsId(criteria.getOrganizationId()),
+                projectId(criteria.getProjectId()),
+                documentInstanceId(criteria.getDocumentInstanceId()),
+                documentInstanceIds(criteria.getDocumentInstanceIds())
         );
     }
 
-    private static Predicate organizationId(String organizationId){
-        if (organizationId == null) {
+    private static Predicate organizationsId(String organizationId) {
+        if (StringUtils.isNullOrEmpty(organizationId)) {
             return null;
         }
-        return documentInstance.organizationId.eq(UUID.fromString(organizationId));
+        return qDocumentInstance.organizationId.eq(UUID.fromString(organizationId));
+
+    }
+
+    private static Predicate projectId(String projectId) {
+        if (StringUtils.isNullOrEmpty(projectId)) {
+            return null;
+        }
+
+        // Assuming you have a relationship between DocumentInstance and ProjectDocument
+        return qDocumentInstance
+                .id.in(
```

Comment:
I have already implemented this in branch `feature-48-extend-repository-findAll-criteria-custom...` but it seems that it's not merged.